src/from_substrait.cpp

```diff
@@ -492,22 +492,59 @@ shared_ptr<Relation> SubstraitToDuckDB::TransformFilterOp(const substrait::Rel &
 	return make_shared_ptr<FilterRelation>(TransformOp(sfilter.input()), TransformExpr(sfilter.condition()));
 }
 
+const google::protobuf::RepeatedField<int32_t>& GetOutputMapping(const substrait::Rel &sop) {
+	const substrait::RelCommon* common = nullptr;
+	switch (sop.rel_type_case()) {
+	case substrait::Rel::RelTypeCase::kJoin:
+		common = &sop.join().common();
+		break;
+	case substrait::Rel::RelTypeCase::kProject:
+		common = &sop.project().common();
+		break;
+	default:
+		throw InternalException("Unsupported relation type " + to_string(sop.rel_type_case()));
+	}
+	if (!common->has_emit()) {
+		static google::protobuf::RepeatedField<int32_t> empty_mapping;
+		return empty_mapping;
+	}
+	return common->emit().output_mapping();
+}
+
 shared_ptr<Relation>
 SubstraitToDuckDB::TransformProjectOp(const substrait::Rel &sop,
                                       const google::protobuf::RepeatedPtrField<std::string> *names) {
 	vector<unique_ptr<ParsedExpression>> expressions;
 	RootNameIterator iterator(names);
 
-	for (auto &sexpr : sop.project().expressions()) {
-		expressions.push_back(TransformExpr(sexpr, &iterator));
+	auto input_rel = TransformOp(sop.project().input());
+
+	auto mapping = GetOutputMapping(sop);
+	auto num_input_columns = input_rel->Columns().size();
+	if (mapping.empty()) {
+		for (int i = 1; i <= num_input_columns; i++) {
```

Comment:
The first column is numbered at zero.  I found https://substrait.io/tutorial/sql_to_substrait/#field-indices to be useful (in addition to the individual relation pages).

Reply:
The index `i` is used to create the duckdb column reference **not** substrait. For reference please see [here](https://github.com/substrait-io/duckdb-substrait-extension/blob/acbd4ab61f0ff6756551f73dcafe3ab7f2394dc6/src/from_substrait.cpp#L181-L181).